From: Petr Tesarik <ptesarik@suse.com>
Date: Tue, 25 Sep 2018 09:23:04 +0200
Subject: Turn off NUMA in the kdump kernel
References: bsc#1109784, bsc#1102609
Upstream: merged
Git-commit: 4c0e78b9923f0dbba3ad1bacc2336d91f7b520e5

On NUMA systems, the crashkernel reserved region may get allocated
on a different node than the low region (memory below 4G). With NUMA
optimization turned on, the kernel will prefer node-local pages for
all page allocations, which may unnecessarily deplete the DMA and
DMA32 regions with data that could be allocated from the Normal
zone (which is on a remote node).

It makes little sense to address this corner case in the memory
management system, because nobody would build a machine with almost
all memory on a remote node. So, instead, let's turn off NUMA for
kdump and accept the potential performance hit.

Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 init/load.sh |    1 +
 1 file changed, 1 insertion(+)

--- a/init/load.sh
+++ b/init/load.sh
@@ -73,6 +73,7 @@ function build_kdump_commandline()
         fi
         # Use deadline for saving the memory footprint
         commandline="$commandline elevator=deadline sysrq=yes reset_devices acpi_no_memhotplug cgroup_disable=memory nokaslr"
+        commandline="$commandline numa=off"
         commandline="$commandline irqpoll ${nr_cpus}"
         commandline="$commandline root=kdump rootflags=bind rd.udev.children-max=8"
         case $(uname -i) in
